tests/sentry/snuba/test_discover.py

```diff
@@ -2871,16 +2871,16 @@ def test_conditions_with_timestamps(self):
         for t, ev in enumerate(events):
             val = ev[0] * 32
             for i in range(ev[1]):
-                data = load_data("transaction", timestamp=before_now(seconds=3 * t + 1))
+                data = load_data("transaction", timestamp=self.now - timedelta(seconds=3 * t + 1))
                 data["transaction"] = f"{val}"
                 self.store_event(data=data, project_id=self.project.id)
 
         for use_snql in [False, True]:
             results = discover.query(
                 selected_columns=["transaction", "count()"],
                 query="event.type:transaction AND (timestamp:<{} OR timestamp:>{})".format(
-                    iso_format(before_now(seconds=5)),
-                    iso_format(before_now(seconds=3)),
+                    iso_format(self.now - timedelta(seconds=5)),
+                    iso_format(self.now - timedelta(seconds=3)),
```

Comment:
This test started failing once I added the `test_comparison_aggregate_function*` tests, but it only failed when running the entire test suite.

I don't completely understand why, since this test it completely unrelated. My best guess is that this test is named `test_conditions_with_timestamps`, so the new tests would be run before this test, and somehow that causes problems with the dates here.

To fix it, I use `self.now` to calculate the dates of data. This seems to resolve the issue, since we use `self.now` as the end date of the query as well.